File: integration-tests/hibernate-orm-panache/src/test/java/io/quarkus/it/panache/DDLGenerationPMT.java
package io.quarkus.it.panache;

import static org.assertj.core.api.Assertions.assertThat;
import static org.hamcrest.Matchers.is;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.RegisterExtension;

import io.quarkus.builder.Version;
import io.quarkus.test.ProdBuildResults;
import io.quarkus.test.ProdModeTestResults;
import io.quarkus.test.QuarkusProdModeTest;
import io.restassured.RestAssured;

/**
 * Verifies that DDL scripts are generated when script generation is configured in application.properties.
 */
public class DDLGenerationPMT {

    @RegisterExtension
    static final QuarkusProdModeTest config = new QuarkusProdModeTest()
            .withApplicationRoot((jar) -> jar
                    .addClasses(PageItem.class, TestResources.class, NoPagingTestEndpoint.class))
            .setApplicationName("ddl-generation")
            .setApplicationVersion(Version.getVersion())
            .setRun(true)
            .setLogFileName("ddl-generation-test.log")
            .withConfigurationResource("ddlgeneration.properties");

    @ProdBuildResults
    private ProdModeTestResults prodModeTestResults;

    @Test
    public void test() {
        RestAssured.when().get("/no-paging-test").then().body(is("OK"));

        assertThat(prodModeTestResults.getBuildDir().resolve("quarkus-app/create.sql").toFile()).exists();
        assertThat(prodModeTestResults.getBuildDir().resolve("quarkus-app/drop.sql").toFile()).exists();
    }

}

File: integration-tests/hibernate-reactive-panache-kotlin/src/test/kotlin/io/quarkus/it/panache/reactive/kotlin/DDLGenerationPMT.kt
package io.quarkus.it.panache.reactive.kotlin

import io.quarkus.builder.Version
import io.quarkus.test.ProdBuildResults
import io.quarkus.test.ProdModeTestResults
import io.quarkus.test.QuarkusProdModeTest
import io.restassured.RestAssured
import org.assertj.core.api.Assertions
import org.hamcrest.Matchers
import org.jboss.shrinkwrap.api.spec.JavaArchive
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.RegisterExtension

/**
 * Verifies that DDL scripts are generated when script generation is configured in
 * application.properties.
 */
class DDLGenerationPMT {
    @ProdBuildResults var prodModeTestResults: ProdModeTestResults? = null

    @Test
    fun test() {
        RestAssured.`when`()["/no-paging-test"].then().body(Matchers.`is`("OK"))
        Assertions.assertThat(
                prodModeTestResults!!.buildDir.resolve("quarkus-app/create.sql").toFile()
            )
            .exists()
        Assertions.assertThat(
                prodModeTestResults!!.buildDir.resolve("quarkus-app/drop.sql").toFile()
            )
            .exists()
    }

    companion object {
        @RegisterExtension
        val config =
            QuarkusProdModeTest()
                .withApplicationRoot { jar: JavaArchive ->
                    jar.addClasses(PageItem::class.java, NoPagingTestEndpoint::class.java)
                }
                .setApplicationName("ddl-generation")
                .setApplicationVersion(Version.getVersion())
                .setRun(true)
                .setLogFileName("ddl-generation-test.log")
                .withConfigurationResource("ddlgeneration.properties")
    }
}
